pytroll_collectors/scisys.py

```diff
@@ -520,15 +538,7 @@ def receive_from_zmq(host, port, station, environment, excluded_platforms,
             logger.debug("to_send: %s", str(to_send))
             if to_send is None:
                 continue
-            if topic_postfix is not None:
-                subject = "/".join(("", to_send['format'],
-                                   to_send['data_processing_level'],
-                                   topic_postfix))
-            else:
-                subject = "/".join(("", to_send['format'],
-                                   to_send['data_processing_level'],
-                                   station, environment,
-                                   "polar", "direct_readout"))
+            subject = get_subject_from_msg2send(to_send, station, environment, topic_postfix)
```

Comment:
This is still untested. The only case now tested is when `to_send` is `None`.

Reply:
Well, yes, correct, the whole `with Publish`was untested. But now I introduced (re-factored) a function to create the adequate subject (`get_subject_from_msg2send`) that I believe is tested for various use-cases, not only when `to_send` is `None`. Or am I mistaken?

Reply:
Codecov is telling that `scisys.py` line 542 is untested. I guess the new tests call `get_subject_from_msg2send()` and thus not going this route and so the line is not covered. The rest of the `receive_from_zmq()` function, including the publishing, is also uncovered due to this.